---
title: "Easy access, interactive exploration and analysis of public gene expression datasets with Phantasus"
author: 
    - name: Alexey Sergushichev
    - email: alsergbox@gmail.com
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{An Example Workshop}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Easy access, interactive exploration and analysis of public gene expression datasets with Phantasus

Author: Alexey Sergushichev.


## Overview

Phantasus is a web tool designed for visual and interactive gene expression analysis. In particular, it was designed to allow to go from a typical dataset to differential expression and downstream analysis in an easy and streamlined manner. For that aim, Phantasus integrates an intuitive JavaScript heatmap interface with gene expression analysis tools from Bioconductor. Phantasus can be used at its official mirror <https://alserglab.wustl.edu/phantasus> or install locally as an R package or a docker image. 

In this vignette, we will go over the basic setup of Phantasus as an R package and its functionality.
Then we will explore several helpful functions of a companion phantasusLite R package
to load RNA-seq counts for NCBI GEO datasets from R environment.

More documentation on how to install and use Phantasus is available at <https://ctlab.github.io/phantasus-doc/>.

## _R_ / _Bioconductor_ packages used

List of _R_ / _Bioconductor_ packages that we will use:

```{r message=FALSE}
library(phantasus)
library(phantasusLite)
library(GEOquery)
library(limma)
```

## Working with local installation of Phantasus 

Before starting a Phantasus server from R we need to configure it. 
For this we can use interactive `setupPhantasus()` function. 
It will create a configuration file and suggest to download files with gene annotations and pathways.

```{r}
setupPhantasus()
```

Let's explore config a bit. The most important options that might be edited are:
1) `host` and `port` to control where the server will be listening, and
2) `preloaded_dir` pointing to a directory with the preloaded datasets (more on that later).
More detailed description of the configuration file is available at <https://ctlab.github.io/phantasus-doc/installation.html#configuration>.

```{r}
configFile <- file.path(tools::R_user_dir(package = "phantasus", 
                                           which = "config"), "user.conf")
readLines(configFile) |> writeLines()
```

Now we can start a server with `servePhantasus()` function. 
For simplicity we use a separate R session, starting it from the RStudio's terminal.
In the command we specify `preloaded` option, as it's not set in the config file.

```{r}
preloadedDir <- file.path(getwd(), "preloaded")
dir.create(preloadedDir)
serveCommand <- sprintf('R --vanilla -e "phantasus::servePhantasus(preloaded=\'%s\')"', preloadedDir)
message("Using the following shell command:")
message(serveCommand)
```

Actually executing the above command with the help of RStudio's API:
```{r eval=FALSE}
serverTerminal <- rstudioapi::terminalExecute(serveCommand)
```

After the server has started we can open Phantasus in the browser:

```{r eval=FALSE}
browseURL("http://localhost:8000/")
```

Please refer to <https://ctlab.github.io/phantasus-doc/quick-start.html> for the vignette
describing Phantasus functionality on the example of microarray dataset GSE53986.
**Note:** the official mirror <https://alserglab.wustl.edu/phantasus/> can be used
in case of problems setting up Phantasus in Bioconductor Workshop environment.

Phantasus can also supports RNA-seq datasets, for which it relies on data from ARCHS4 and DEE2 projects.
For example, we can open RNA-seq dataset GSE53053 by passing directly in the URL:

```{r eval=FALSE}
browseURL("http://localhost:8000/phantasus/index.html?geo=GSE53053")
```

A vignette describing analysis of RNA-seq dataset GSE53053 can be found at <https://ctlab.github.io/phantasus-doc/tutorials/GSE53053/>. 

A particular benefit of setting up Phantasus server is the ability to share
preprocessed datasets. For example, let consider the dataset GSE53986
which we can load and process using R:

```{r}
gse53986 <- getGEO("GSE53986", AnnotGPL = TRUE)[[1]]
gse53986$condition <- gse53986$`treatment:ch1`
pData(gse53986) <- pData(gse53986)[, c("title", "geo_accession", "condition")]

fData(gse53986) <- fData(gse53986)[, c("Gene ID", "Gene symbol")]
exprs(gse53986) <- normalizeBetweenArrays(log2(exprs(gse53986)), method="quantile")
```

We can save it as an RDA file into the preloaded directory: 

```{r}
save(gse53986, file=file.path(preloadedDir, "GSE53986_norm.rda"))
```

Now we can open it using URL like this (with removed file extension):

```{r eval=FALSE}
browseURL("http://localhost:8000/phantasus/index.html?preloaded=GSE53986_norm")
```

Further, Phantasus can generate a session link with _File/Get session link..._, which 
allows to reopen or share the session, preserving the visuals (color scheme, sorting, etc.).
Such links can be used to generate preloaded datasets as well, which would also preserve the visuals.
**Note:** session links generated from Phantasus instance within Bioconductor Workshop environment
will not work here due to security restrictions.

```{r}
sessionURL <- "https://alserglab.wustl.edu/phantasus/?session=x00000000000000" # example session URL
generatePreloadedSession(sessionURL, preloadedName = "my_session", preloadedDir = preloadedDir)
```

This dataset now can be opened using the same-style URL as before:

```{r eval=FALSE}
browseURL("http://localhost:8000/phantasus/index.html?preloaded=my_session")
```

After finishing working with Phantasus we can kill the process
with Ctrl-C, or via RStudio API:
```{r eval=FALSE}
rstudioapi::terminalKill(serverTerminal)
```

## Loading precomputed RNA-seq counts for GEO datasets in R

In this section we will explore how to use `phantasusLite` package to easily
load precomputed RNA-seq matrices for GEO datasets. 

First, let's load dataset GSE53053 from GEO using `GEOquery` package:

```{r message=FALSE}
ess <- getGEO("GSE53053")
es <- ess[[1]]
```

GSE53053 is an RNA-seq dataset, and GEO database does not store RNA-seq expression matrices
in a standard format (although it's changing: https://www.ncbi.nlm.nih.gov/geo/info/rnaseqcounts.html).
Thus `exprs(es)` is empty:

```{r}
head(exprs(es))
```

However, projects like ARCHS4 and DEE2 provide count data for many RNA-seq samples,
mostly for human and mouse, but also for other organisms.
We made their data available for an easy remote access by setting up an HSDS server
'<https://alserglab.wustl.edu/hsds/>'. It allows remote access to HDF5 files with counts
from these projects:

```{r}
hsdsURL <- 'https://alserglab.wustl.edu/hsds/?domain=/counts'
getHSDSFileList(hsdsURL)
```

GSE53053 dataset was sequenced from *Mus musculus* and we can get an expression matrix
from the corresponding HDF5-file with DEE2 data:

```{r}
file <- "dee2/mmusculus_star_matrix_20240409.h5"
es <- loadCountsFromH5FileHSDS(es, hsdsURL, file)
head(exprs(es))
```

Instead of specifying the particular file for `loadCountsFromH5FileHSDS`,
function `loadCountsFromHSDS` can be used to automatically select the HDF5-file
with the largest number of quantified samples:

```{r}
es <- ess[[1]]
es <- loadCountsFromHSDS(es, hsdsURL)
head(exprs(es))
```

The counts are different from the previous values as ARCHS4 counts has been used:
ARCHS4 is prioritized when there are several files with the same number of samples.
The used file name can be obtained from the `experimentData` slot of the `ExpressionSet` object:

```{r}
preproc(experimentData(es))$gene_counts_source
```

Further, gene symbols are also imported from the database and are available as feature data:
```{r}
head(fData(es))
```

## Session info

```{r}
sessionInfo()
```
