---
title: "Easy access, interactive exploration and analysis of public gene expression datasets with Phantasus"
author: 
    - name: Alexey Sergushichev
    - email: alsergbox@gmail.com
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{An Example Workshop}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Easy access, interactive exploration and analysis of public gene expression datasets with Phantasus

Author: Alexey Sergushichev.


## Overview

Phantasus is a web tool designed for visual and interactive gene expression analysis. In particular, it was designed to allow to go from a typical dataset to differential expression and downstream analysis in an easy and streamlined manner. For that aim, Phantasus integrates an intuitive JavaScript heatmap interface with gene expression analysis tools from Bioconductor. Phantasus can be used at its official mirror <https://alserglab.wustl.edu/phantasus> or install locally as an R package or a docker image. 

In this vignette, we will go over the basic setup of Phantasus as an R package and its functionality.
Then we will explore several helpful functions of a companion phantasusLite R package
to load RNA-seq counts for NCBI GEO datasets from R environment.

More documentation on how to install and use Phantasus is available at <https://ctlab.github.io/phantasus-doc/>.

## _R_ / _Bioconductor_ packages used

List of _R_ / _Bioconductor_ packages that we will use:

```{r message=FALSE}
library(phantasus)
library(phantasusLite)
library(GEOquery)
library(limma)
```

## Phantasus setup

Before starting a Phantasus server from R we need to configure it. 
For this we can use interactive `setupPhantasus()` function. 
It will create a configuration file and suggest to download files with gene annotations and pathways.

```{r}
setupPhantasus()
```

Let's explore config a bit. The most important options that might be edited are:
1) `host` and `port` to control where the server will be listening, and
2) `preloaded_dir` pointing to a directory with the preloaded datasets (more on that later).
More detailed description of the configuration file is available at <https://ctlab.github.io/phantasus-doc/installation.html#configuration>.

```{r}
configFile <- file.path(tools::R_user_dir(package = "phantasus", 
                                           which = "config"), "user.conf")
readLines(configFile) |> writeLines()
```

Now we can start a server with `servePhantasus()` function. 
Usually it can be ran with the default arguments, however for this vignette 
we will set several options: `background=TRUE` makes server start in the background,
`openInBrowser=FALSE` prevents automatic opening of the browser with Phantasus.
Preloaded directory is recommended to be set by editing the configuration
file, but also can be set via `preloaded` option.

```{r}
preloadedDir <- file.path(getwd(), "preloaded")
dir.create(preloadedDir)
server <- servePhantasus(background=TRUE, openInBrowser = FALSE, preloadedDir = preloadedDir)
```

After server has started we can open Phantasus in the browser:

```{r eval=FALSE}
browseURL("http://localhost:8000/")
```

Please refer to <https://ctlab.github.io/phantasus-doc/quick-start.html> for the vignette
describing Phantasus functionality on the example of microarray dataset GSE53986.

Phantasus can also supports RNA-seq datasets, for which it relies on data from ARCHS4 and DEE2 projects.
For example, we can open RNA-seq dataset GSE53053 by passing directly in the URL:

```{r eval=FALSE}
browseURL("http://localhost:8000/phantasus/index.html?geo=GSE53053")
```

A vignette describing analysis of RNA-seq dataset GSE53053 can be found at <https://ctlab.github.io/phantasus-doc/tutorials/GSE53053/>.

A particular benefit of setting up Phantasus server is the ability to share
preprocessed datasets. For example, let consider the dataset GSE14308
which we can load and process using R:

```{r}
gse14308 <- getGEO("GSE14308", AnnotGPL = TRUE)[[1]]
gse14308$condition <- sub("-.*$", "", gse14308$title)
pData(gse14308) <- pData(gse14308)[, c("title", "geo_accession", "condition")]
gse14308 <- gse14308[, order(gse14308$condition)]

fData(gse14308) <- fData(gse14308)[, c("Gene ID", "Gene symbol")]
exprs(gse14308) <- normalizeBetweenArrays(log2(exprs(gse14308)+1), method="quantile")
```

We can save it as an RDA file into the preloaded folder: 

```{r}
save(gse14308, file=file.path(preloadedDir, "GSE14308_norm.rda"))
```

Now we can open it using URL like this (with removed file extension):

```{r eval=FALSE}
browseURL("http://localhost:8000/phantasus/index.html?preloaded=GSE14308_norm")
```

After we are done working we can stop the server:

```{r}
server$stop()
```

## Loading precomputed RNA-seq counts for GEO datasets in R

Let's load dataset GSE53053 from GEO using `GEOquery` package:

```{r message=FALSE}
ess <- getGEO("GSE53053")
es <- ess[[1]]
```

RNA-seq dataset from GEO do not contain the expression matrix, thus `exprs(es)` is empty:

```{r}
head(exprs(es))
```

However, a number of precomputed gene count tables are available at HSDS server  '<https://alserglab.wustl.edu/hsds/>'. It features HDF5 files with counts
from ARCHS4 and DEE2 projects:

```{r}
url <- 'https://alserglab.wustl.edu/hsds/?domain=/counts'
getHSDSFileList(url)
```

GSE53053 dataset was sequenced from *Mus musculus* and we can get an expression matrix
from the corresponding HDF5-file with DEE2 data:

```{r}
file <- "dee2/mmusculus_star_matrix_20240409.h5"
es <- loadCountsFromH5FileHSDS(es, url, file)
head(exprs(es))
```

Normally `loadCountsFromHSDS` can be used to automatically select the HDF5-file with the largest
number of quantified samples:

```{r}
es <- ess[[1]]
es <- loadCountsFromHSDS(es, url)
head(exprs(es))
```

The counts are different from the previous values as ARCHS4 counts were used -- ARCHS4 is prioritized when there are several files with the same number of samples:

```{r}
preproc(experimentData(es))$gene_counts_source
```

Further, gene symbols are also imported from ARCHS4 database and are available as feature data:
```{r}
head(fData(es))
```
