<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Easy access, interactive exploration and analysis of public gene expression datasets with Phantasus â€¢ phantasusPkgDemo</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Easy access, interactive exploration and analysis of public gene expression datasets with Phantasus">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">phantasusPkgDemo</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/phantasusPkgDemo.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/assaron/phantasusPkgDemo" class="external-link">
    <span class="fa fa-github"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Easy access, interactive exploration and
analysis of public gene expression datasets with Phantasus</h1>
                        <h4 data-toc-skip class="author">Alexey
Sergushichev</h4>
                                    <h4 data-toc-skip class="author">true</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/assaron/phantasusPkgDemo/blob/devel/vignettes/phantasusPkgDemo.Rmd" class="external-link"><code>vignettes/phantasusPkgDemo.Rmd</code></a></small>
      <div class="hidden name"><code>phantasusPkgDemo.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="easy-access-interactive-exploration-and-analysis-of-public-gene-expression-datasets-with-phantasus">Easy access, interactive exploration and analysis of public gene
expression datasets with Phantasus<a class="anchor" aria-label="anchor" href="#easy-access-interactive-exploration-and-analysis-of-public-gene-expression-datasets-with-phantasus"></a>
</h2>
<p>Author: Alexey Sergushichev.</p>
<div class="section level3">
<h3 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h3>
<p>Phantasus is a web tool designed for visual and interactive gene
expression analysis. In particular, it was designed to allow to go from
a typical dataset to differential expression and downstream analysis in
an easy and streamlined manner. For that aim, Phantasus integrates an
intuitive JavaScript heatmap interface with gene expression analysis
tools from Bioconductor. Phantasus can be used at its official mirror <a href="https://alserglab.wustl.edu/phantasus" class="external-link uri">https://alserglab.wustl.edu/phantasus</a> or install locally
as an R package or a docker image.</p>
<p>In this vignette, we will go over the basic setup of Phantasus as an
R package and its functionality. Then we will explore several helpful
functions of a companion phantasusLite R package to load RNA-seq counts
for NCBI GEO datasets from R environment.</p>
<p>More documentation on how to install and use Phantasus is available
at <a href="https://ctlab.github.io/phantasus-doc/" class="external-link uri">https://ctlab.github.io/phantasus-doc/</a>.</p>
</div>
<div class="section level3">
<h3 id="r-bioconductor-packages-used">
<em>R</em> / <em>Bioconductor</em> packages used<a class="anchor" aria-label="anchor" href="#r-bioconductor-packages-used"></a>
</h3>
<p>List of <em>R</em> / <em>Bioconductor</em> packages that we will
use:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://alserglab.wustl.edu/phantasus" class="external-link">phantasus</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ctlab/phantasusLite/" class="external-link">phantasusLite</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/seandavi/GEOquery" class="external-link">GEOquery</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioinf.wehi.edu.au/limma/" class="external-link">limma</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="working-with-local-installation-of-phantasus">Working with local installation of Phantasus<a class="anchor" aria-label="anchor" href="#working-with-local-installation-of-phantasus"></a>
</h3>
<p>Before starting a Phantasus server from R we need to configure it.
For this we can use interactive <code><a href="https://rdrr.io/pkg/phantasus/man/setupPhantasus.html" class="external-link">setupPhantasus()</a></code> function.
It will create a configuration file and suggest to download files with
gene annotations and pathways.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/phantasus/man/setupPhantasus.html" class="external-link">setupPhantasus</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Use default setup parameters</span></span>
<span><span class="co">#&gt; Use existed configuration file: /github/home/.config/R/phantasus/user.conf</span></span>
<span><span class="co">#&gt; Current configuration:  default</span></span>
<span><span class="co">#&gt; Configure GEO cache folder...</span></span>
<span><span class="co">#&gt; ! GEO cache folder exists and is treated as already configured !</span></span>
<span><span class="co">#&gt; Configure AnnotationDb...</span></span>
<span><span class="co">#&gt; ! AnnotationDB folder exists and is treated as already configured !</span></span>
<span><span class="co">#&gt; Configure FGSEA pathways...</span></span>
<span><span class="co">#&gt; ! FGSEA folder exists and is treated as already configured !</span></span>
<span><span class="co">#&gt; Configure RNA-seq counts...</span></span>
<span><span class="co">#&gt; ! RNA-seq counts path looks like web URL !</span></span>
<span><span class="co">#&gt; https://alserglab.wustl.edu/hsds/?domain=/counts response as HSDS server</span></span>
<span><span class="co">#&gt; HSDS server will be used as source of RNA-seq counts</span></span>
<span><span class="co">#&gt; ! Successfully setup configuration file: /github/home/.config/R/phantasus/user.conf !</span></span></code></pre></div>
<p>Letâ€™s explore config a bit. The most important options that might be
edited are: 1) <code>host</code> and <code>port</code> to control where
the server will be listening, and 2) <code>preloaded_dir</code> pointing
to a directory with the preloaded datasets (more on that later). More
detailed description of the configuration file is available at <a href="https://ctlab.github.io/phantasus-doc/installation.html#configuration" class="external-link uri">https://ctlab.github.io/phantasus-doc/installation.html#configuration</a>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">configFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu">tools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/tools/userdir.html" class="external-link">R_user_dir</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"phantasus"</span>, </span>
<span>                                           which <span class="op">=</span> <span class="st">"config"</span><span class="op">)</span>, <span class="st">"user.conf"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/readLines.html" class="external-link">readLines</a></span><span class="op">(</span><span class="va">configFile</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/writeLines.html" class="external-link">writeLines</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; default:</span></span>
<span><span class="co">#&gt;   host: "0.0.0.0"</span></span>
<span><span class="co">#&gt;   port: 8000</span></span>
<span><span class="co">#&gt;   preloaded_dir: NULL</span></span>
<span><span class="co">#&gt;   static_root: !expr system.file('www/phantasus.js', package = 'phantasus')</span></span>
<span><span class="co">#&gt;   cache_root: "/github/home/.cache/R/phantasus"</span></span>
<span><span class="co">#&gt;   cache_folders:</span></span>
<span><span class="co">#&gt;     geo_path: !expr file.path(cache_root, 'geo/')</span></span>
<span><span class="co">#&gt;     annot_db: !expr file.path(cache_root, 'annotationdb/')</span></span>
<span><span class="co">#&gt;     fgsea_pathways: !expr file.path(cache_root, 'fgsea_pathways/')</span></span>
<span><span class="co">#&gt;     rnaseq_counts: https://alserglab.wustl.edu/hsds/?domain=/counts</span></span>
<span><span class="co">#&gt;   geo_mirrors:</span></span>
<span><span class="co">#&gt;     true_geo: "https://ftp.ncbi.nlm.nih.gov"</span></span>
<span><span class="co">#&gt;     minimal_cache: "https://alserglab.wustl.edu/files/phantasus/minimal-cache/"</span></span></code></pre></div>
<p>Now we can start a server with <code><a href="https://rdrr.io/pkg/phantasus/man/servePhantasus.html" class="external-link">servePhantasus()</a></code>
function. For simplicity we use a separate R session, starting it from
the RStudioâ€™s terminal. In the command we specify <code>preloaded</code>
option, as itâ€™s not set in the config file.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">preloadedDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"preloaded"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">preloadedDir</span><span class="op">)</span></span>
<span><span class="va">serveCommand</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">'R --vanilla -e "phantasus::servePhantasus(preloaded=\'%s\')"'</span>, <span class="va">preloadedDir</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"Using the following shell command:"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using the following shell command:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="va">serveCommand</span><span class="op">)</span></span>
<span><span class="co">#&gt; R --vanilla -e "phantasus::servePhantasus(preloaded='/__w/phantasusPkgDemo/phantasusPkgDemo/vignettes/preloaded')"</span></span></code></pre></div>
<p>Actually executing the above command with the help of RStudioâ€™s
API:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">serverTerminal</span> <span class="op">&lt;-</span> <span class="fu">rstudioapi</span><span class="fu">::</span><span class="fu"><a href="https://rstudio.github.io/rstudioapi/reference/terminalExecute.html" class="external-link">terminalExecute</a></span><span class="op">(</span><span class="va">serveCommand</span><span class="op">)</span></span></code></pre></div>
<p>After the server has started we can open Phantasus in the
browser:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/browseURL.html" class="external-link">browseURL</a></span><span class="op">(</span><span class="st">"http://localhost:8000/"</span><span class="op">)</span></span></code></pre></div>
<p>Please refer to <a href="https://ctlab.github.io/phantasus-doc/quick-start.html" class="external-link uri">https://ctlab.github.io/phantasus-doc/quick-start.html</a>
for the vignette describing Phantasus functionality on the example of
microarray dataset GSE53986. <strong>Note:</strong> the official mirror
<a href="https://alserglab.wustl.edu/phantasus/" class="external-link uri">https://alserglab.wustl.edu/phantasus/</a> can be used in
case of problems setting up Phantasus in Bioconductor Workshop
environment.</p>
<p>Phantasus can also supports RNA-seq datasets, for which it relies on
data from ARCHS4 and DEE2 projects. For example, we can open RNA-seq
dataset GSE53053 by passing directly in the URL:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/browseURL.html" class="external-link">browseURL</a></span><span class="op">(</span><span class="st">"http://localhost:8000/phantasus/index.html?geo=GSE53053"</span><span class="op">)</span></span></code></pre></div>
<p>A vignette describing analysis of RNA-seq dataset GSE53053 can be
found at <a href="https://ctlab.github.io/phantasus-doc/tutorials/GSE53053/" class="external-link uri">https://ctlab.github.io/phantasus-doc/tutorials/GSE53053/</a>.</p>
<p>A particular benefit of setting up Phantasus server is the ability to
share preprocessed datasets. For example, let consider the dataset
GSE53986 which we can load and process using R:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse53986</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://seandavi.github.io/GEOquery/reference/getGEO.html" class="external-link">getGEO</a></span><span class="op">(</span><span class="st">"GSE53986"</span>, AnnotGPL <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; Found 1 file(s)</span></span>
<span><span class="co">#&gt; GSE53986_series_matrix.txt.gz</span></span>
<span><span class="va">gse53986</span><span class="op">$</span><span class="va">condition</span> <span class="op">&lt;-</span> <span class="va">gse53986</span><span class="op">$</span><span class="va">`treatment:ch1`</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">gse53986</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">gse53986</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"title"</span>, <span class="st">"geo_accession"</span>, <span class="st">"condition"</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">gse53986</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">gse53986</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Gene ID"</span>, <span class="st">"Gene symbol"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">gse53986</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/normalizebetweenarrays.html" class="external-link">normalizeBetweenArrays</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">gse53986</span><span class="op">)</span><span class="op">)</span>, method<span class="op">=</span><span class="st">"quantile"</span><span class="op">)</span></span></code></pre></div>
<p>We can save it as an RDA file into the preloaded directory:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/save.html" class="external-link">save</a></span><span class="op">(</span><span class="va">gse53986</span>, file<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">preloadedDir</span>, <span class="st">"GSE53986_norm.rda"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Now we can open it using URL like this (with removed file
extension):</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/browseURL.html" class="external-link">browseURL</a></span><span class="op">(</span><span class="st">"http://localhost:8000/phantasus/index.html?preloaded=GSE53986_norm"</span><span class="op">)</span></span></code></pre></div>
<p>Further, Phantasus can generate a session link with <em>File/Get
session linkâ€¦</em>, which allows to reopen or share the session,
preserving the visuals (color scheme, sorting, etc.). Such links can be
used to generate preloaded datasets as well, which would also preserve
the visuals.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sessionURL</span> <span class="op">&lt;-</span> <span class="st">"https://alserglab.wustl.edu/phantasus/?session=x00000000000000"</span> <span class="co"># use your own URL here</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/phantasus/man/generatePreloadedSession.html" class="external-link">generatePreloadedSession</a></span><span class="op">(</span><span class="va">sessionURL</span>, preloadedName <span class="op">=</span> <span class="st">"my_session"</span>, preloadedDir <span class="op">=</span> <span class="va">preloadedDir</span><span class="op">)</span></span></code></pre></div>
<p>This dataset now can be opened using the same-style URL as
before:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/browseURL.html" class="external-link">browseURL</a></span><span class="op">(</span><span class="st">"http://localhost:8000/phantasus/index.html?preloaded=my_session"</span><span class="op">)</span></span></code></pre></div>
<p>After finishing working with Phantasus we can kill the process with
Ctrl-C, or via RStudio API:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rstudioapi</span><span class="fu">::</span><span class="fu"><a href="https://rstudio.github.io/rstudioapi/reference/terminalKill.html" class="external-link">terminalKill</a></span><span class="op">(</span><span class="va">serverTerminal</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="loading-precomputed-rna-seq-counts-for-geo-datasets-in-r">Loading precomputed RNA-seq counts for GEO datasets in R<a class="anchor" aria-label="anchor" href="#loading-precomputed-rna-seq-counts-for-geo-datasets-in-r"></a>
</h3>
<p>In this section we will explore how to use <code>phantasusLite</code>
package to easily load precomputed RNA-seq matrices for GEO
datasets.</p>
<p>First, letâ€™s load dataset GSE53053 from GEO using
<code>GEOquery</code> package:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ess</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://seandavi.github.io/GEOquery/reference/getGEO.html" class="external-link">getGEO</a></span><span class="op">(</span><span class="st">"GSE53053"</span><span class="op">)</span></span>
<span><span class="va">es</span> <span class="op">&lt;-</span> <span class="va">ess</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p>GSE53053 is an RNA-seq dataset, and GEO database does not store
RNA-seq expression matrices in a standard format (although itâ€™s
changing: <a href="https://www.ncbi.nlm.nih.gov/geo/info/rnaseqcounts.html" class="external-link uri">https://www.ncbi.nlm.nih.gov/geo/info/rnaseqcounts.html</a>).
Thus <code>exprs(es)</code> is empty:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;      GSM1281300 GSM1281301 GSM1281302 GSM1281303 GSM1281304 GSM1281305</span></span>
<span><span class="co">#&gt;      GSM1281306 GSM1281307</span></span></code></pre></div>
<p>However, projects like ARCHS4 and DEE2 provide count data for many
RNA-seq samples, mostly for human and mouse, but also for other
organisms. We made their data available for an easy remote access by
setting up an HSDS server â€˜<a href="https://alserglab.wustl.edu/hsds/" class="external-link uri">https://alserglab.wustl.edu/hsds/</a>â€™. It allows remote
access to HDF5 files with counts from these projects:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hsdsURL</span> <span class="op">&lt;-</span> <span class="st">'https://alserglab.wustl.edu/hsds/?domain=/counts'</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/phantasusLite/man/getHSDSFileList.html" class="external-link">getHSDSFileList</a></span><span class="op">(</span><span class="va">hsdsURL</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "/counts/archs4/Arabidopsis_thaliana_count_matrix.h5"    </span></span>
<span><span class="co">#&gt;  [2] "/counts/archs4/Bos_taurus_count_matrix.h5"              </span></span>
<span><span class="co">#&gt;  [3] "/counts/archs4/Caenorhabditis_elegans_count_matrix.h5"  </span></span>
<span><span class="co">#&gt;  [4] "/counts/archs4/Danio_rerio_count_matrix.h5"             </span></span>
<span><span class="co">#&gt;  [5] "/counts/archs4/Drosophila_melanogaster_count_matrix.h5" </span></span>
<span><span class="co">#&gt;  [6] "/counts/archs4/Gallus_gallus_count_matrix.h5"           </span></span>
<span><span class="co">#&gt;  [7] "/counts/archs4/Rattus_norvegicus_count_matrix.h5"       </span></span>
<span><span class="co">#&gt;  [8] "/counts/archs4/Saccharomyces_cerevisiae_count_matrix.h5"</span></span>
<span><span class="co">#&gt;  [9] "/counts/archs4/human_gene_v2.3.h5"                      </span></span>
<span><span class="co">#&gt; [10] "/counts/archs4/meta.h5"                                 </span></span>
<span><span class="co">#&gt; [11] "/counts/archs4/mouse_gene_v2.3.h5"                      </span></span>
<span><span class="co">#&gt; [12] "/counts/dee2/athaliana_star_matrix_20240409.h5"         </span></span>
<span><span class="co">#&gt; [13] "/counts/dee2/celegans_star_matrix_20240409.h5"          </span></span>
<span><span class="co">#&gt; [14] "/counts/dee2/dmelanogaster_star_matrix_20240409.h5"     </span></span>
<span><span class="co">#&gt; [15] "/counts/dee2/drerio_star_matrix_20240409.h5"            </span></span>
<span><span class="co">#&gt; [16] "/counts/dee2/ecoli_star_matrix_20240409.h5"             </span></span>
<span><span class="co">#&gt; [17] "/counts/dee2/hsapiens_star_matrix_20240409.h5"          </span></span>
<span><span class="co">#&gt; [18] "/counts/dee2/meta.h5"                                   </span></span>
<span><span class="co">#&gt; [19] "/counts/dee2/mmusculus_star_matrix_20240409.h5"         </span></span>
<span><span class="co">#&gt; [20] "/counts/dee2/rnorvegicus_star_matrix_20240409.h5"       </span></span>
<span><span class="co">#&gt; [21] "/counts/dee2/scerevisiae_star_matrix_20240409.h5"</span></span></code></pre></div>
<p>GSE53053 dataset was sequenced from <em>Mus musculus</em> and we can
get an expression matrix from the corresponding HDF5-file with DEE2
data:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">file</span> <span class="op">&lt;-</span> <span class="st">"dee2/mmusculus_star_matrix_20240409.h5"</span></span>
<span><span class="va">es</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phantasusLite/man/loadCountsFromH5FileHSDS.html" class="external-link">loadCountsFromH5FileHSDS</a></span><span class="op">(</span><span class="va">es</span>, <span class="va">hsdsURL</span>, <span class="va">file</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                    GSM1281300 GSM1281301 GSM1281302 GSM1281303 GSM1281304</span></span>
<span><span class="co">#&gt; ENSMUSG00000102693          0          0          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000064842          0          0          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000051951          0          0        135        120        132</span></span>
<span><span class="co">#&gt; ENSMUSG00000102851          0          0          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000103377          0          0          0          0          1</span></span>
<span><span class="co">#&gt; ENSMUSG00000104017          0          0          0          0          0</span></span>
<span><span class="co">#&gt;                    GSM1281305 GSM1281306 GSM1281307</span></span>
<span><span class="co">#&gt; ENSMUSG00000102693          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000064842          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000051951          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000102851          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000103377          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000104017          0          0          0</span></span></code></pre></div>
<p>Instead of specifying the particular file for
<code>loadCountsFromH5FileHSDS</code>, function
<code>loadCountsFromHSDS</code> can be used to automatically select the
HDF5-file with the largest number of quantified samples:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">es</span> <span class="op">&lt;-</span> <span class="va">ess</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">es</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phantasusLite/man/loadCountsFromHSDS.html" class="external-link">loadCountsFromHSDS</a></span><span class="op">(</span><span class="va">es</span>, <span class="va">hsdsURL</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                    GSM1281300 GSM1281301 GSM1281302 GSM1281303 GSM1281304</span></span>
<span><span class="co">#&gt; ENSMUSG00000000001       1015        603        561        549        425</span></span>
<span><span class="co">#&gt; ENSMUSG00000000003          0          0          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000000028        109         34          0         14          9</span></span>
<span><span class="co">#&gt; ENSMUSG00000000031          0         18          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000000037          0          0          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000000049          0          0          0          0          0</span></span>
<span><span class="co">#&gt;                    GSM1281305 GSM1281306 GSM1281307</span></span>
<span><span class="co">#&gt; ENSMUSG00000000001        853        407        479</span></span>
<span><span class="co">#&gt; ENSMUSG00000000003          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000000028        165          0         15</span></span>
<span><span class="co">#&gt; ENSMUSG00000000031          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000000037          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000000049          0          0          0</span></span></code></pre></div>
<p>The counts are different from the previous values as ARCHS4 counts
has been used: ARCHS4 is prioritized when there are several files with
the same number of samples. The used file name can be obtained from the
<code>experimentData</code> slot of the <code>ExpressionSet</code>
object:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/class.MIAME.html" class="external-link">preproc</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/abstract.html" class="external-link">experimentData</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">gene_counts_source</span></span>
<span><span class="co">#&gt; [1] "archs4/mouse_gene_v2.3.h5"</span></span></code></pre></div>
<p>Further, gene symbols are also imported from the database and are
available as feature data:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                    Gene symbol          ENSEMBLID</span></span>
<span><span class="co">#&gt; ENSMUSG00000000001       Gnai3 ENSMUSG00000000001</span></span>
<span><span class="co">#&gt; ENSMUSG00000000003        Pbsn ENSMUSG00000000003</span></span>
<span><span class="co">#&gt; ENSMUSG00000000028       Cdc45 ENSMUSG00000000028</span></span>
<span><span class="co">#&gt; ENSMUSG00000000031         H19 ENSMUSG00000000031</span></span>
<span><span class="co">#&gt; ENSMUSG00000000037       Scml2 ENSMUSG00000000037</span></span>
<span><span class="co">#&gt; ENSMUSG00000000049        Apoh ENSMUSG00000000049</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h3>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Etc/UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] limma_3.61.4        GEOquery_2.73.0     Biobase_2.65.0     </span></span>
<span><span class="co">#&gt; [4] BiocGenerics_0.51.0 phantasusLite_1.3.0 phantasus_1.25.2   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] DBI_1.2.3               opencpu_2.2.12          rlang_1.1.4            </span></span>
<span><span class="co">#&gt;   [4] magrittr_2.0.3          matrixStats_1.3.0       compiler_4.4.1         </span></span>
<span><span class="co">#&gt;   [7] RSQLite_2.3.7           png_0.1-8               systemfonts_1.1.0      </span></span>
<span><span class="co">#&gt;  [10] vctrs_0.6.5             pkgconfig_2.0.3         crayon_1.5.3           </span></span>
<span><span class="co">#&gt;  [13] fastmap_1.2.0           XVector_0.45.0          utf8_1.2.4             </span></span>
<span><span class="co">#&gt;  [16] promises_1.3.0          rmarkdown_2.27          tzdb_0.4.0             </span></span>
<span><span class="co">#&gt;  [19] ccaPP_0.3.3             UCSC.utils_1.1.0        ragg_1.3.2             </span></span>
<span><span class="co">#&gt;  [22] purrr_1.0.2             bit_4.0.5               xfun_0.46              </span></span>
<span><span class="co">#&gt;  [25] zlibbioc_1.51.1         cachem_1.1.0            GenomeInfoDb_1.41.1    </span></span>
<span><span class="co">#&gt;  [28] jsonlite_1.8.8          blob_1.2.4              later_1.3.2            </span></span>
<span><span class="co">#&gt;  [31] DelayedArray_0.31.9     rhdf5filters_1.17.0     Rhdf5lib_1.27.0        </span></span>
<span><span class="co">#&gt;  [34] BiocParallel_1.39.0     parallel_4.4.1          R6_2.5.1               </span></span>
<span><span class="co">#&gt;  [37] bslib_0.7.0             jquerylib_0.1.4         Rcpp_1.0.13            </span></span>
<span><span class="co">#&gt;  [40] knitr_1.48              R.utils_2.12.3          readr_2.1.5            </span></span>
<span><span class="co">#&gt;  [43] IRanges_2.39.2          httpuv_1.6.15           Matrix_1.7-0           </span></span>
<span><span class="co">#&gt;  [46] tidyselect_1.2.1        abind_1.4-5             yaml_2.3.9             </span></span>
<span><span class="co">#&gt;  [49] codetools_0.2-20        curl_5.2.1              lattice_0.22-6         </span></span>
<span><span class="co">#&gt;  [52] tibble_3.2.1            webutils_1.2.0          withr_3.0.0            </span></span>
<span><span class="co">#&gt;  [55] KEGGREST_1.45.1         askpass_1.2.0           evaluate_0.24.0        </span></span>
<span><span class="co">#&gt;  [58] desc_1.4.3              xml2_1.3.6              Biostrings_2.73.1      </span></span>
<span><span class="co">#&gt;  [61] pillar_1.9.0            MatrixGenerics_1.17.0   stats4_4.4.1           </span></span>
<span><span class="co">#&gt;  [64] pcaPP_2.0-4             generics_0.1.3          S4Vectors_0.43.2       </span></span>
<span><span class="co">#&gt;  [67] hms_1.1.3               ggplot2_3.5.1           munsell_0.5.1          </span></span>
<span><span class="co">#&gt;  [70] scales_1.3.0            glue_1.7.0              tools_4.4.1            </span></span>
<span><span class="co">#&gt;  [73] robustbase_0.99-3       data.table_1.15.4       fgsea_1.31.0           </span></span>
<span><span class="co">#&gt;  [76] locfit_1.5-9.10         fs_1.6.4                mvtnorm_1.2-5          </span></span>
<span><span class="co">#&gt;  [79] fastmatch_1.1-4         cowplot_1.1.3           rhdf5_2.49.0           </span></span>
<span><span class="co">#&gt;  [82] grid_4.4.1              tidyr_1.3.1             AnnotationDbi_1.67.0   </span></span>
<span><span class="co">#&gt;  [85] edgeR_4.3.5             colorspace_2.1-0        GenomeInfoDbData_1.2.12</span></span>
<span><span class="co">#&gt;  [88] config_0.3.2            cli_3.6.3               brew_1.0-10            </span></span>
<span><span class="co">#&gt;  [91] Rook_1.2                textshaping_0.4.0       rhdf5client_1.27.1     </span></span>
<span><span class="co">#&gt;  [94] fansi_1.0.6             S4Arrays_1.5.5          svglite_2.1.3          </span></span>
<span><span class="co">#&gt;  [97] dplyr_1.1.4             gtable_0.3.5            DEoptimR_1.1-3         </span></span>
<span><span class="co">#&gt; [100] R.methodsS3_1.8.2       sass_0.4.9              digest_0.6.36          </span></span>
<span><span class="co">#&gt; [103] SparseArray_1.5.25      rjson_0.2.21            htmlwidgets_1.6.4      </span></span>
<span><span class="co">#&gt; [106] R.oo_1.26.0             memoise_2.0.1           htmltools_0.5.8.1      </span></span>
<span><span class="co">#&gt; [109] pkgdown_2.1.0           lifecycle_1.0.4         httr_1.4.7             </span></span>
<span><span class="co">#&gt; [112] statmod_1.5.0           mime_0.12               openssl_2.2.0          </span></span>
<span><span class="co">#&gt; [115] bit64_4.0.5</span></span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Alexey Sergushichev.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
