<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Easy access, interactive exploration and analysis of public gene expression datasets with Phantasus • phantasusPkgDemo</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Easy access, interactive exploration and analysis of public gene expression datasets with Phantasus">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">phantasusPkgDemo</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/phantasusPkgDemo.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/assaron/phantasusPkgDemo" class="external-link">
    <span class="fa fa-github"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Easy access, interactive exploration and
analysis of public gene expression datasets with Phantasus</h1>
                        <h4 data-toc-skip class="author">Alexey
Sergushichev</h4>
                                    <h4 data-toc-skip class="author">true</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/assaron/phantasusPkgDemo/blob/devel/vignettes/phantasusPkgDemo.Rmd" class="external-link"><code>vignettes/phantasusPkgDemo.Rmd</code></a></small>
      <div class="hidden name"><code>phantasusPkgDemo.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="easy-access-interactive-exploration-and-analysis-of-public-gene-expression-datasets-with-phantasus">Easy access, interactive exploration and analysis of public gene
expression datasets with Phantasus<a class="anchor" aria-label="anchor" href="#easy-access-interactive-exploration-and-analysis-of-public-gene-expression-datasets-with-phantasus"></a>
</h2>
<p>Author: Alexey Sergushichev.</p>
<div class="section level3">
<h3 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h3>
<p>Phantasus is a web tool designed for visual and interactive gene
expression analysis. In particular, it was designed to allow to go from
a typical dataset to differential expression and downstream analysis in
an easy and streamlined manner. For that aim, Phantasus integrates an
intuitive JavaScript heatmap interface with gene expression analysis
tools from Bioconductor. Phantasus can be used at its official mirror <a href="https://alserglab.wustl.edu/phantasus" class="external-link uri">https://alserglab.wustl.edu/phantasus</a> or install locally
as an R package or a docker image.</p>
<p>In this vignette, we will go over the basic setup of Phantasus as an
R package and its functionality. Then we will explore several helpful
functions of a companion phantasusLite R package to load RNA-seq counts
for NCBI GEO datasets from R environment.</p>
<p>More documentation on how to install and use Phantasus is available
at <a href="https://ctlab.github.io/phantasus-doc/" class="external-link uri">https://ctlab.github.io/phantasus-doc/</a>.</p>
</div>
<div class="section level3">
<h3 id="r-bioconductor-packages-used">
<em>R</em> / <em>Bioconductor</em> packages used<a class="anchor" aria-label="anchor" href="#r-bioconductor-packages-used"></a>
</h3>
<p>List of <em>R</em> / <em>Bioconductor</em> packages that we will
use:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://alserglab.wustl.edu/phantasus" class="external-link">phantasus</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ctlab/phantasusLite/" class="external-link">phantasusLite</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/seandavi/GEOquery" class="external-link">GEOquery</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioinf.wehi.edu.au/limma/" class="external-link">limma</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="phantasus-setup">Phantasus setup<a class="anchor" aria-label="anchor" href="#phantasus-setup"></a>
</h3>
<p>Before starting a Phantasus server from R we need to configure it.
For this we can use interactive <code><a href="https://rdrr.io/pkg/phantasus/man/setupPhantasus.html" class="external-link">setupPhantasus()</a></code> function.
It will create a configuration file and suggest to download files with
gene annotations and pathways.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/phantasus/man/setupPhantasus.html" class="external-link">setupPhantasus</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Use default setup parameters</span></span>
<span><span class="co">#&gt; Use existed configuration file: /github/home/.config/R/phantasus/user.conf</span></span>
<span><span class="co">#&gt; Current configuration:  default</span></span>
<span><span class="co">#&gt; Configure GEO cache folder...</span></span>
<span><span class="co">#&gt; ! GEO cache folder exists and is treated as already configured !</span></span>
<span><span class="co">#&gt; Configure AnnotationDb...</span></span>
<span><span class="co">#&gt; ! AnnotationDB folder exists and is treated as already configured !</span></span>
<span><span class="co">#&gt; Configure FGSEA pathways...</span></span>
<span><span class="co">#&gt; ! FGSEA folder exists and is treated as already configured !</span></span>
<span><span class="co">#&gt; Configure RNA-seq counts...</span></span>
<span><span class="co">#&gt; ! RNA-seq counts path looks like web URL !</span></span>
<span><span class="co">#&gt; https://alserglab.wustl.edu/hsds/?domain=/counts response as HSDS server</span></span>
<span><span class="co">#&gt; HSDS server will be used as source of RNA-seq counts</span></span>
<span><span class="co">#&gt; ! Successfully setup configuration file: /github/home/.config/R/phantasus/user.conf !</span></span></code></pre></div>
<p>Let’s explore config a bit. The most important options that might be
edited are: 1) <code>host</code> and <code>port</code> to control where
the server will be listening, and 2) <code>preloaded_dir</code> pointing
to a directory with the preloaded datasets (more on that later). More
detailed description of the configuration file is available at <a href="https://ctlab.github.io/phantasus-doc/installation.html#configuration" class="external-link uri">https://ctlab.github.io/phantasus-doc/installation.html#configuration</a>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">configFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu">tools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/tools/userdir.html" class="external-link">R_user_dir</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"phantasus"</span>, </span>
<span>                                           which <span class="op">=</span> <span class="st">"config"</span><span class="op">)</span>, <span class="st">"user.conf"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/readLines.html" class="external-link">readLines</a></span><span class="op">(</span><span class="va">configFile</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/writeLines.html" class="external-link">writeLines</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; default:</span></span>
<span><span class="co">#&gt;   host: "0.0.0.0"</span></span>
<span><span class="co">#&gt;   port: 8000</span></span>
<span><span class="co">#&gt;   preloaded_dir: NULL</span></span>
<span><span class="co">#&gt;   static_root: !expr system.file('www/phantasus.js', package = 'phantasus')</span></span>
<span><span class="co">#&gt;   cache_root: "/github/home/.cache/R/phantasus"</span></span>
<span><span class="co">#&gt;   cache_folders:</span></span>
<span><span class="co">#&gt;     geo_path: !expr file.path(cache_root, 'geo/')</span></span>
<span><span class="co">#&gt;     annot_db: !expr file.path(cache_root, 'annotationdb/')</span></span>
<span><span class="co">#&gt;     fgsea_pathways: !expr file.path(cache_root, 'fgsea_pathways/')</span></span>
<span><span class="co">#&gt;     rnaseq_counts: https://alserglab.wustl.edu/hsds/?domain=/counts</span></span>
<span><span class="co">#&gt;   geo_mirrors:</span></span>
<span><span class="co">#&gt;     true_geo: "https://ftp.ncbi.nlm.nih.gov"</span></span>
<span><span class="co">#&gt;     minimal_cache: "https://alserglab.wustl.edu/files/phantasus/minimal-cache/"</span></span></code></pre></div>
<p>Now we can start a server with <code><a href="https://rdrr.io/pkg/phantasus/man/servePhantasus.html" class="external-link">servePhantasus()</a></code>
function. Usually it can be ran with the default arguments, however for
this vignette we will set several options: <code>background=TRUE</code>
makes server start in the background, <code>openInBrowser=FALSE</code>
prevents automatic opening of the browser with Phantasus. Preloaded
directory is recommended to be set by editing the configuration file,
but also can be set via <code>preloaded</code> option.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">preloadedDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"preloaded"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">preloadedDir</span><span class="op">)</span></span>
<span><span class="va">server</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phantasus/man/servePhantasus.html" class="external-link">servePhantasus</a></span><span class="op">(</span>background<span class="op">=</span><span class="cn">TRUE</span>, openInBrowser <span class="op">=</span> <span class="cn">FALSE</span>, preloadedDir <span class="op">=</span> <span class="va">preloadedDir</span><span class="op">)</span></span>
<span><span class="co">#&gt; Preloaded dir is not set</span></span>
<span><span class="co">#&gt; 31 counts files are available</span></span>
<span><span class="co">#&gt; 2 annotationDb are available</span></span>
<span><span class="co">#&gt; 6 fgsea tables are available</span></span>
<span><span class="co">#&gt; Server was started with following parameters: host=0.0.0.0, port=8000</span></span></code></pre></div>
<p>After server has started we can open Phantasus in the browser:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/browseURL.html" class="external-link">browseURL</a></span><span class="op">(</span><span class="st">"http://localhost:8000/"</span><span class="op">)</span></span></code></pre></div>
<p>Please refer to <a href="https://ctlab.github.io/phantasus-doc/quick-start.html" class="external-link uri">https://ctlab.github.io/phantasus-doc/quick-start.html</a>
for the vignette describing Phantasus functionality on the example of
microarray dataset GSE53986.</p>
<p>Phantasus can also supports RNA-seq datasets, for which it relies on
data from ARCHS4 and DEE2 projects. For example, we can open RNA-seq
dataset GSE53053 by passing directly in the URL:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/browseURL.html" class="external-link">browseURL</a></span><span class="op">(</span><span class="st">"http://localhost:8000/phantasus/index.html?geo=GSE53053"</span><span class="op">)</span></span></code></pre></div>
<p>A vignette describing analysis of RNA-seq dataset GSE53053 can be
found at <a href="https://ctlab.github.io/phantasus-doc/tutorials/GSE53053/" class="external-link uri">https://ctlab.github.io/phantasus-doc/tutorials/GSE53053/</a>.</p>
<p>A particular benefit of setting up Phantasus server is the ability to
share preprocessed datasets. For example, let consider the dataset
GSE14308 which we can load and process using R:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse14308</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://seandavi.github.io/GEOquery/reference/getGEO.html" class="external-link">getGEO</a></span><span class="op">(</span><span class="st">"GSE14308"</span>, AnnotGPL <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; Found 1 file(s)</span></span>
<span><span class="co">#&gt; GSE14308_series_matrix.txt.gz</span></span>
<span><span class="va">gse14308</span><span class="op">$</span><span class="va">condition</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"-.*$"</span>, <span class="st">""</span>, <span class="va">gse14308</span><span class="op">$</span><span class="va">title</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">gse14308</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">gse14308</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"title"</span>, <span class="st">"geo_accession"</span>, <span class="st">"condition"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">gse14308</span> <span class="op">&lt;-</span> <span class="va">gse14308</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">gse14308</span><span class="op">$</span><span class="va">condition</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">gse14308</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">gse14308</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Gene ID"</span>, <span class="st">"Gene symbol"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">gse14308</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/normalizebetweenarrays.html" class="external-link">normalizeBetweenArrays</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">gse14308</span><span class="op">)</span><span class="op">+</span><span class="fl">1</span><span class="op">)</span>, method<span class="op">=</span><span class="st">"quantile"</span><span class="op">)</span></span></code></pre></div>
<p>We can save it as an RDA file into the preloaded folder:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/save.html" class="external-link">save</a></span><span class="op">(</span><span class="va">gse14308</span>, file<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">preloadedDir</span>, <span class="st">"GSE14308_norm.rda"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Now we can open it using URL like this (with removed file
extension):</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/browseURL.html" class="external-link">browseURL</a></span><span class="op">(</span><span class="st">"http://localhost:8000/phantasus/index.html?preloaded=GSE14308_norm"</span><span class="op">)</span></span></code></pre></div>
<p>After we are done working we can stop the server:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">server</span><span class="op">$</span><span class="kw">stop</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="loading-precomputed-rna-seq-counts-for-geo-datasets-in-r">Loading precomputed RNA-seq counts for GEO datasets in R<a class="anchor" aria-label="anchor" href="#loading-precomputed-rna-seq-counts-for-geo-datasets-in-r"></a>
</h3>
<p>Let’s load dataset GSE53053 from GEO using <code>GEOquery</code>
package:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ess</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://seandavi.github.io/GEOquery/reference/getGEO.html" class="external-link">getGEO</a></span><span class="op">(</span><span class="st">"GSE53053"</span><span class="op">)</span></span>
<span><span class="va">es</span> <span class="op">&lt;-</span> <span class="va">ess</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p>RNA-seq dataset from GEO do not contain the expression matrix, thus
<code>exprs(es)</code> is empty:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;      GSM1281300 GSM1281301 GSM1281302 GSM1281303 GSM1281304 GSM1281305</span></span>
<span><span class="co">#&gt;      GSM1281306 GSM1281307</span></span></code></pre></div>
<p>However, a number of precomputed gene count tables are available at
HSDS server ‘<a href="https://alserglab.wustl.edu/hsds/" class="external-link uri">https://alserglab.wustl.edu/hsds/</a>’. It features HDF5
files with counts from ARCHS4 and DEE2 projects:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">url</span> <span class="op">&lt;-</span> <span class="st">'https://alserglab.wustl.edu/hsds/?domain=/counts'</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/phantasusLite/man/getHSDSFileList.html" class="external-link">getHSDSFileList</a></span><span class="op">(</span><span class="va">url</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "/counts/archs4/Arabidopsis_thaliana_count_matrix.h5"    </span></span>
<span><span class="co">#&gt;  [2] "/counts/archs4/Bos_taurus_count_matrix.h5"              </span></span>
<span><span class="co">#&gt;  [3] "/counts/archs4/Caenorhabditis_elegans_count_matrix.h5"  </span></span>
<span><span class="co">#&gt;  [4] "/counts/archs4/Danio_rerio_count_matrix.h5"             </span></span>
<span><span class="co">#&gt;  [5] "/counts/archs4/Drosophila_melanogaster_count_matrix.h5" </span></span>
<span><span class="co">#&gt;  [6] "/counts/archs4/Gallus_gallus_count_matrix.h5"           </span></span>
<span><span class="co">#&gt;  [7] "/counts/archs4/Rattus_norvegicus_count_matrix.h5"       </span></span>
<span><span class="co">#&gt;  [8] "/counts/archs4/Saccharomyces_cerevisiae_count_matrix.h5"</span></span>
<span><span class="co">#&gt;  [9] "/counts/archs4/human_gene_v2.3.h5"                      </span></span>
<span><span class="co">#&gt; [10] "/counts/archs4/meta.h5"                                 </span></span>
<span><span class="co">#&gt; [11] "/counts/archs4/mouse_gene_v2.3.h5"                      </span></span>
<span><span class="co">#&gt; [12] "/counts/dee2/athaliana_star_matrix_20240409.h5"         </span></span>
<span><span class="co">#&gt; [13] "/counts/dee2/celegans_star_matrix_20240409.h5"          </span></span>
<span><span class="co">#&gt; [14] "/counts/dee2/dmelanogaster_star_matrix_20240409.h5"     </span></span>
<span><span class="co">#&gt; [15] "/counts/dee2/drerio_star_matrix_20240409.h5"            </span></span>
<span><span class="co">#&gt; [16] "/counts/dee2/ecoli_star_matrix_20240409.h5"             </span></span>
<span><span class="co">#&gt; [17] "/counts/dee2/hsapiens_star_matrix_20240409.h5"          </span></span>
<span><span class="co">#&gt; [18] "/counts/dee2/meta.h5"                                   </span></span>
<span><span class="co">#&gt; [19] "/counts/dee2/mmusculus_star_matrix_20240409.h5"         </span></span>
<span><span class="co">#&gt; [20] "/counts/dee2/rnorvegicus_star_matrix_20240409.h5"       </span></span>
<span><span class="co">#&gt; [21] "/counts/dee2/scerevisiae_star_matrix_20240409.h5"       </span></span>
<span><span class="co">#&gt; [22] "/counts/dee2_old/athaliana_star_matrix_20221107.h5"     </span></span>
<span><span class="co">#&gt; [23] "/counts/dee2_old/celegans_star_matrix_20221107.h5"      </span></span>
<span><span class="co">#&gt; [24] "/counts/dee2_old/dmelanogaster_star_matrix_20221107.h5" </span></span>
<span><span class="co">#&gt; [25] "/counts/dee2_old/drerio_star_matrix_20211102.h5"        </span></span>
<span><span class="co">#&gt; [26] "/counts/dee2_old/ecoli_star_matrix_20221107.h5"         </span></span>
<span><span class="co">#&gt; [27] "/counts/dee2_old/hsapiens_star_matrix_20221107.h5"      </span></span>
<span><span class="co">#&gt; [28] "/counts/dee2_old/meta.h5"                               </span></span>
<span><span class="co">#&gt; [29] "/counts/dee2_old/mmusculus_star_matrix_20221107.h5"     </span></span>
<span><span class="co">#&gt; [30] "/counts/dee2_old/rnorvegicus_star_matrix_20221107.h5"   </span></span>
<span><span class="co">#&gt; [31] "/counts/dee2_old/scerevisiae_star_matrix_20221107.h5"</span></span></code></pre></div>
<p>GSE53053 dataset was sequenced from <em>Mus musculus</em> and we can
get an expression matrix from the corresponding HDF5-file with DEE2
data:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">file</span> <span class="op">&lt;-</span> <span class="st">"dee2/mmusculus_star_matrix_20240409.h5"</span></span>
<span><span class="va">es</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phantasusLite/man/loadCountsFromH5FileHSDS.html" class="external-link">loadCountsFromH5FileHSDS</a></span><span class="op">(</span><span class="va">es</span>, <span class="va">url</span>, <span class="va">file</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                    GSM1281300 GSM1281301 GSM1281302 GSM1281303 GSM1281304</span></span>
<span><span class="co">#&gt; ENSMUSG00000102693          0          0          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000064842          0          0          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000051951          0          0        135        120        132</span></span>
<span><span class="co">#&gt; ENSMUSG00000102851          0          0          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000103377          0          0          0          0          1</span></span>
<span><span class="co">#&gt; ENSMUSG00000104017          0          0          0          0          0</span></span>
<span><span class="co">#&gt;                    GSM1281305 GSM1281306 GSM1281307</span></span>
<span><span class="co">#&gt; ENSMUSG00000102693          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000064842          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000051951          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000102851          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000103377          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000104017          0          0          0</span></span></code></pre></div>
<p>Normally <code>loadCountsFromHSDS</code> can be used to automatically
select the HDF5-file with the largest number of quantified samples:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">es</span> <span class="op">&lt;-</span> <span class="va">ess</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">es</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phantasusLite/man/loadCountsFromHSDS.html" class="external-link">loadCountsFromHSDS</a></span><span class="op">(</span><span class="va">es</span>, <span class="va">url</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                    GSM1281300 GSM1281301 GSM1281302 GSM1281303 GSM1281304</span></span>
<span><span class="co">#&gt; ENSMUSG00000000001       1015        603        561        549        425</span></span>
<span><span class="co">#&gt; ENSMUSG00000000003          0          0          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000000028        109         34          0         14          9</span></span>
<span><span class="co">#&gt; ENSMUSG00000000031          0         18          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000000037          0          0          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000000049          0          0          0          0          0</span></span>
<span><span class="co">#&gt;                    GSM1281305 GSM1281306 GSM1281307</span></span>
<span><span class="co">#&gt; ENSMUSG00000000001        853        407        479</span></span>
<span><span class="co">#&gt; ENSMUSG00000000003          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000000028        165          0         15</span></span>
<span><span class="co">#&gt; ENSMUSG00000000031          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000000037          0          0          0</span></span>
<span><span class="co">#&gt; ENSMUSG00000000049          0          0          0</span></span></code></pre></div>
<p>The counts are different from the previous values as ARCHS4 counts
were used – ARCHS4 is prioritized when there are several files with the
same number of samples:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/class.MIAME.html" class="external-link">preproc</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/abstract.html" class="external-link">experimentData</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">gene_counts_source</span></span>
<span><span class="co">#&gt; [1] "archs4/mouse_gene_v2.3.h5"</span></span></code></pre></div>
<p>Further, gene symbols are also imported from ARCHS4 database and are
available as feature data:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">es</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                    Gene symbol          ENSEMBLID</span></span>
<span><span class="co">#&gt; ENSMUSG00000000001       Gnai3 ENSMUSG00000000001</span></span>
<span><span class="co">#&gt; ENSMUSG00000000003        Pbsn ENSMUSG00000000003</span></span>
<span><span class="co">#&gt; ENSMUSG00000000028       Cdc45 ENSMUSG00000000028</span></span>
<span><span class="co">#&gt; ENSMUSG00000000031         H19 ENSMUSG00000000031</span></span>
<span><span class="co">#&gt; ENSMUSG00000000037       Scml2 ENSMUSG00000000037</span></span>
<span><span class="co">#&gt; ENSMUSG00000000049        Apoh ENSMUSG00000000049</span></span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Alexey Sergushichev.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
